peer/brontide_test.go

```diff
@@ -1315,7 +1315,6 @@ func TestHandleRemovePendingChannel(t *testing.T) {
 		}
 
 		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
```

Comment:
IIUC, the problem is that all the test cases share the same `peer` object, which causes problems not only with  parallel execution, but also potentially with serial execution if new test cases are added or the order of them changes.

Shouldn't we create a new peer object for each test cases instead?

Reply:
Good call. Updated as suggested.